taglibs/async/await-tag.js

```diff
@@ -23,6 +23,10 @@ function promiseToCallback(promise, callback, thisObj) {
             finalPromise = finalPromise.fail(function(err) {
                 callback(err);
             });
+        } else {
+            finalPromise = finalPromise.then(void 0, function(err) {
```

Comment:
As defined by the Promises/A+ specification (either explicitly or implicitly, I can't remember which), `onResolved` and `onRejected` are simply ignored if they're not a function. The only concern would be if a function was assigned to `window.undefined`, which wouldn't be possible in IE9+ or other modern browsers.

I do agree that `undefined`/`null` is easier to read. I would be in favor of `null`, as long as it wouldn't break any of the popular libraries (which it shouldn't).